### PR TITLE
remove some deprecated SpanData methods

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java [hunk 1]
```diff
@@ -228,17 +215,6 @@ static Link create(SpanContext spanContext, Attributes attributes, int totalAttr
...
     SpanContext getSpanContext();
 
-    /**
-     * Returns the {@code SpanContext}.
-     *
-     * @return the {@code SpanContext}.
-     * @deprecated Use {@link #getSpanContext()}
-     */
-    @Deprecated
-    default SpanContext getContext() {
```
Comment (bogdandrutu) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java:238]: Should we deprecate the other accessors for ids in favor of this one?

It has already helpers for bytes for ids and we can remove the helpers to convert.

Comment (jkwatson) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java:238]: This one was deprecated in favor of `getSpanContext()`, so I think we want to delete this one, either way.

Comment (bogdandrutu) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java:238]: Good point :)

**Review by bogdandrutu — APPROVED** [review]

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java:238]: This is in the link not in the SpanData :)

**Pull request merged**